lib/typhoeus/response/header.rb

```diff
@@ -47,7 +47,7 @@ def parse
       # @return [ void ]
       def process_line(header)
         key, value = header.split(':', 2)
-        process_pair(key.strip, value.strip.gsub(/\r?\n\s*/, ' '))
+        process_pair(key.strip, (value ? value.strip.gsub(/\r?\n\s*/, ' ') : nil))
```

Comment:
I don't have a strong opinion on this, returning `nil` works and allows to distinguish between a blank value or malformed header, but I'm totally ok with returning an empty string.

`Net:HTTP` raises and closes the connexion:
```
Net::HTTPBadResponse: wrong header line format
	from /home/bigbourin/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http/response.rb:63:in `each_response_header'
	from /home/bigbourin/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http/response.rb:31:in `read_new'
	from /home/bigbourin/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http.rb:1437:in `block in transport_request'
	from /home/bigbourin/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http.rb:1434:in `catch'
	from /home/bigbourin/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http.rb:1434:in `transport_request'
	from /home/bigbourin/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http.rb:1407:in `request'
	from /home/bigbourin/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http.rb:1308:in `request_get'
	from /home/bigbourin/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http.rb:481:in `block in get_response'
	from /home/bigbourin/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http.rb:853:in `start'
	from /home/bigbourin/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http.rb:584:in `start'
	from /home/bigbourin/.rbenv/versions/2.3.0/lib/ruby/2.3.0/net/http.rb:479:in `get_response'
```

Which is slightly better than Typhoeus (explicit error) but not very good IMO, HTTP is supposed to be resilient and this is a very easy to dodge parsing error. I tried with google chrome for example and it parses without any issue, it just ignores the invalid header.

We could also ignore the line altogether but to me it feels like it's the responsibility of the caller to do that, not the low level. If I want to write and HTTP header validator for example, I need the most accurate representation as possible.